File: java/scf-simple/src/main/java/org/mydotey/scf/type/string/StringToDoubleConverter.java
package org.mydotey.scf.type.string;

/**
 * @author koqizhao
 *
 * May 21, 2018
 */
public class StringToDoubleConverter extends StringConverter<Double> {

    public static final StringToDoubleConverter DEFAULT = new StringToDoubleConverter();

    public StringToDoubleConverter() {
        super(Double.class);
    }

    @Override
    public Double convert(String source) {
        return Double.parseDouble(source);
    }

}

File: dotnet/src/MyDotey.SCF.Simple/Type/String/StringToDoubleConverter.cs
using System;

namespace MyDotey.SCF.Type.String
{
    /**
     * @author koqizhao
     *
     * May 21, 2018
     */
    public class StringToDoubleConverter : StringConverter<double?>
    {
        public static readonly StringToDoubleConverter DEFAULT = new StringToDoubleConverter();

        public override double? Convert(string source)
        {
            if (string.IsNullOrWhiteSpace(source))
                return null;

            return double.Parse(source);
        }
    }
}